main.ts

```diff
@@ -28,6 +28,7 @@ export default class MyPlugin extends Plugin {
     this.toggl = new TogglService(this);
     if (this.settings.apiToken != null || this.settings.apiToken != "") {
       this.toggl.refreshApiConnection(this.settings.apiToken);
+      this.toggl.setRefreshInterval(this.settings.autoRefreshInterval);
```

Comment:
Since `this.toggl.refreshApiConnection` is async, I think it would make sense to `await` the successful initial setup of the API connection before setting the refresh interval.

Perhaps setting the refresh interval can be done at the end of `refreshApiConnection` instead? Then it's covered by every current (and future) code path that refreshes the connection or changes the API token. 

Currently it's only set on plugin load and encloses the API token set at that time as the token used to refresh.

Reply:
Hmm, an `await` here also blocks the command registration and other view initializations below it.

Merged the code paths, but now `refreshApiConnection` sets an interval just to destroy it one cycle later. Should work, but doesn't feel like the best way to do it.